src/Games/NexusMods.Games.StardewValley/StardewValleyLoadoutSynchronizer.cs

```diff
@@ -0,0 +1,94 @@
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.Loadouts.Mods;
+using NexusMods.Abstractions.Loadouts.Synchronizers;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.MnemonicDB.Abstractions.Models;
+using NexusMods.Paths;
+using File = NexusMods.Abstractions.Loadouts.Files.File;
+
+namespace NexusMods.Games.StardewValley;
+
+public class StardewValleyLoadoutSynchronizer : ALoadoutSynchronizer
+{
+    public StardewValleyLoadoutSynchronizer(IServiceProvider provider) : base(provider) { }
+
+
+    protected override async Task<Loadout.Model> AddChangedFilesToLoadout(Loadout.Model loadout, TempEntity[] newFiles)
+    {
+        using var tx = Connection.BeginTransaction();
+        var overridesMod = GetOrCreateOverridesMod(loadout, tx);
+
+        var smapiModDirectoryNameToModel = new Dictionary<RelativePath, Mod.Model>();
+
+        foreach (var newFile in newFiles)
+        {
+            newFile.Add(File.Loadout, loadout.Id);
+
+            if (!newFile.Contains(File.To))
+            {
+                AddToOverride(newFile);
+                continue;
+            }
+
+            var gamePath = newFile.GetFirst(File.To);
+            if (!IsModFile(gamePath, out var modDirectoryName))
+            {
+                AddToOverride(newFile);
+                continue;
+            }
+
+            if (!smapiModDirectoryNameToModel.TryGetValue(modDirectoryName, out var smapiMod))
+            {
+                smapiMod = GetSMAPIMod(modDirectoryName, loadout, loadout.Db);
+                if (smapiMod is null)
+                {
+                    AddToOverride(newFile);
+                    continue;
+                }
+
+                smapiModDirectoryNameToModel[modDirectoryName] = smapiMod;
+            }
+
+            newFile.Add(File.Mod, smapiMod.Id);
+            newFile.AddTo(tx);
+        }
+
+        var result = await tx.Commit();
+        return result.Db.Get<Loadout.Model>(loadout.Id);
+
+        void AddToOverride(TempEntity newFile)
+        {
+            newFile.Add(File.Mod, overridesMod.Id);
+            newFile.AddTo(tx);
+        }
+    }
+
+    private static Mod.Model? GetSMAPIMod(RelativePath modDirectoryName, Loadout.Model loadout, IDb db)
+    {
+        var manifestFilePath = new GamePath(LocationId.Game, Constants.ModsFolder.Join(modDirectoryName).Join(Constants.ManifestFile));
+
+        var manifestFile = db
+            .Find(File.To)
```

Comment:
The syntax here is odd, but if this is `.FindIndexed(manifestFilePath, File.To)` that will return only files where `To == manifestFilePath`. That should get rid of the need to scan everything. Still not optimal we'll get some better query logic for this at some poitn.

Reply:
I thought `FindIndexed` only works if the attribute has been marked with `IsIndexed`, which `File.To` is not.

Reply:
Ah, we should certainly add IsIndexed to .To then, we'll want that for this attribute.

Reply:
Is this a trivial change? I want to keep this PR focused.

Reply:
Yeah, no big deal we can fix this later